src/hwmon.cpp

```diff
@@ -261,8 +281,10 @@ string HwmonInterface<HwmonT>::lookup()
 			if (found_paths_.size() == 0)
 				throw DriverInitError(path + ": " + "Could not find any hwmons in " + path);
 		}
-		else
-			found_paths_.push_back(path);
+		else {
+			vector<filesystem::path> paths = dir_entries<filter_driver_file>(path);
+			found_paths_.assign(paths.begin(), paths.end());
```

Comment:
```suggestion
			found_paths_.swap(paths);
```
Also why don't we assign `found_paths_` directly in #285?